### ISSUE TITLE
ReactJS RefluxJS separation question

### ISSUE THREAD

**Issue by clintwood** [reflux/refluxjs#380]
I'm totally new to and loving RefluxJS!

I get that the origins of Flux is off the back of ReactJS from FB so the bias is heavily leaning towards using RefluxJS with ReactJS and as such there appears to be a lot of code (I may well be wrong in this assessment) related to mixins for ReactJS.

I'm using the Polymer framework and it would be ideal if I could use RefluxJS 'core' without the ReactJS specific bits since I will probably build Polymer specific mixins.  So the question is, seeing that the Flux pattern is really UI framework agnostic would it not make sense to structure RefluxJS in a way that has a core that is a pure Flux implementation with either an opt-in/out or plugin mechanism for ReactJS specific bits.

Maybe it could be structured like this:
```js
// As is now and must stay: core RefluxJS + ReactJS bits
var Reflux - require('reflux');  // <-- (1)

// but also provide separate modules
var Reflux = require('reflux-core');  // core RefluxJS pure Flux implementation only

//  this is same as (1) above so you wouldn't do this for use with ReactJS, just use (1)
var refluxReact = require('reflux-react');  // ReactJS bits
Reflux.use(refluxReact(...));

// then this becomes possible (in another context)
var refluxPolymer = require('reflux-polymer'); // or Angular etc etc
Reflux.use(refluxPolymer(...));

```
I totally understand that one can just not use the ReactJS bits but feel that not making the ReactJS mandatory may open up RefluxJS to other use cases in other frameworks.

**Comment by LongLiveCHIEF** [comment]
This is most definitely on our roadmap, and we've been very heavily working on core design change concepts these. the last several days.

Whatever the outcome, Reflux will not have "React-ness" in the core, and will in fact be more friendly towards any view library.  We also plan to design in a way to do Reflux/addons, and then create a Reflux React component. (#293)

**Comment by clintwood** [comment]
That is awesome news! Looking forward to that.

**Comment by spoike** [comment]
I think we can do this with the 0.2 branch quite easily (and this has been requested in #107). Will see if we can do it soon.

**Comment by spoike** [comment]
Released reflux-core 0.2.0-rc on npm if you want to play around with it right now. Let me know if there is anything else needed to be done. The repo is at: https://github.com/reflux/reflux-core. It is a fork of refluxjs, but with all the react specifics removed.
